### ISSUE TITLE
AccessDeniedException when starting up after update

### ISSUE THREAD

**Issue by sneakyx** [elastic/elasticsearch#109438]
### Elasticsearch Version

8.14.0

### Java Version

22.0.1

### OS Version

Linux stage 5.15.0-107-generic #117-Ubuntu SMP Fri Apr 26 12:26:49 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

### Problem Description

This morning I updated via `apt update && apt upgrade` elasticsearch from 8.13.4 to 8.14.0 
The update/upgrade didn't show any errors, but when I tried to access the node, I got an error.
When I tried to restart the elasticsearch-service, I got an error and found this in the log file:
```
org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:74) ~[elasticsearch-8.14.0.jar:?]
Caused by: java.io.UncheckedIOException: java.nio.file.AccessDeniedException: /usr/lib/mysql/plugin/auth_pam_tool_dir
	at java.nio.file.FileTreeIterator.fetchNextIfNeeded(FileTreeIterator.java:87) ~[?:?]
```

### Steps to Reproduce

Well, that seems not easy. I just updated/upgraded via `apt update && apt upgrade `

### Logs (if relevant)

```
[2024-06-06T06:10:17,181][WARN ][o.e.n.NativeAccess       ] [stage] Unable to load native provider. Native methods will be disabled.
java.lang.ExceptionInInitializerError: null
	at org.elasticsearch.nativeaccess.lib.NativeLibraryProvider.getLibrary(NativeLibraryProvider.java:58) ~[elasticsearch-native-8.14.0.jar:?]
	at org.elasticsearch.nativeaccess.LinuxNativeAccess.<init>(LinuxNativeAccess.java:20) ~[elasticsearch-native-8.14.0.jar:?]
	at org.elasticsearch.nativeaccess.NativeAccessHolder.<clinit>(NativeAccessHolder.java:28) ~[elasticsearch-native-8.14.0.jar:?]
	at org.elasticsearch.nativeaccess.NativeAccess.instance(NativeAccess.java:22) ~[elasticsearch-native-8.14.0.jar:?]
	at org.elasticsearch.bootstrap.Elasticsearch.initializeNatives(Elasticsearch.java:285) ~[elasticsearch-8.14.0.jar:?]
	at org.elasticsearch.bootstrap.Elasticsearch.initPhase2(Elasticsearch.java:169) ~[elasticsearch-8.14.0.jar:?]
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:74) ~[elasticsearch-8.14.0.jar:?]
Caused by: java.io.UncheckedIOException: java.nio.file.AccessDeniedException: /usr/lib/mysql/plugin/auth_pam_tool_dir
	at java.nio.file.FileTreeIterator.fetchNextIfNeeded(FileTreeIterator.java:87) ~[?:?]
	at java.nio.file.FileTreeIterator.hasNext(FileTreeIterator.java:103) ~[?:?]
	at java.util.Iterator.forEachRemaining(Iterator.java:132) ~[?:?]
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1939) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:556) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:546) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265) ~[?:?]
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:611) ~[?:?]
	at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:291) ~[?:?]
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:194) ~[?:?]
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:212) ~[?:?]
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:1024) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:556) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:546) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:622) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:291) ~[?:?]
	at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:631) ~[?:?]
	at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:637) ~[?:?]
	at java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:642) ~[?:?]
	at org.elasticsearch.nativeaccess.jdk.JdkSystemdLibrary.findLibSystemd(JdkSystemdLibrary.java:74) ~[elasticsearch-native-8.14.0.jar:?]
	at org.elasticsearch.nativeaccess.jdk.JdkSystemdLibrary.<clinit>(JdkSystemdLibrary.java:36) ~[elasticsearch-native-8.14.0.jar:?]
	... 7 more
Caused by: java.nio.file.AccessDeniedException: /usr/lib/mysql/plugin/auth_pam_tool_dir
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:90) ~[?:?]
	at sun.nio.fs.UnixException.asIOException(UnixException.java:115) ~[?:?]
	at sun.nio.fs.UnixFileSystemProvider.newDirectoryStream(UnixFileSystemProvider.java:477) ~[?:?]
	at java.nio.file.Files.newDirectoryStream(Files.java:481) ~[?:?]
	at java.nio.file.FileTreeWalker.visit(FileTreeWalker.java:301) ~[?:?]
	at java.nio.file.FileTreeWalker.next(FileTreeWalker.java:374) ~[?:?]
	at java.nio.file.FileTreeIterator.fetchNextIfNeeded(FileTreeIterator.java:83) ~[?:?]
	at java.nio.file.FileTreeIterator.hasNext(FileTreeIterator.java:103) ~[?:?]
	at java.util.Iterator.forEachRemaining(Iterator.java:132) ~[?:?]
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1939) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:556) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:546) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265) ~[?:?]
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:611) ~[?:?]
	at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:291) ~[?:?]
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:194) ~[?:?]
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:212) ~[?:?]
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:1024) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:556) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:546) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:622) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:291) ~[?:?]
	at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:631) ~[?:?]
	at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:637) ~[?:?]
	at java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:642) ~[?:?]
	at org.elasticsearch.nativeaccess.jdk.JdkSystemdLibrary.findLibSystemd(JdkSystemdLibrary.java:74) ~[elasticsearch-native-8.14.0.jar:?]
	at org.elasticsearch.nativeaccess.jdk.JdkSystemdLibrary.<clinit>(JdkSystemdLibrary.java:36) ~[elasticsearch-native-8.14.0.jar:?]
	... 7 more
[2024-06-06T06:10:17,200][WARN ][o.e.n.NativeAccess       ] [stage] Cannot check if running as root because native access is not available
[2024-06-06T06:10:17,436][INFO ][o.a.l.i.v.PanamaVectorizationProvider] [stage] Java vector incubator API enabled; uses preferredBitSize=256; FMA enabled
[2024-06-06T06:10:17,814][INFO ][o.e.n.Node               ] [stage] version[8.14.0], pid[5275], build[deb/8d96bbe3bf5fed931f3119733895458eab75dca9/2024-06-03T10:05:49.073003402Z], OS[Linux/5.15.0-107-generic/amd64], JVM[Oracle Corporation/OpenJDK 64-Bit Server VM/22.0.1/22.0.1+8-16]
[2024-06-06T06:10:17,815][INFO ][o.e.n.Node               ] [stage] JVM home [/usr/share/elasticsearch/jdk], using bundled JDK [true]
[2024-06-06T06:10:17,815][INFO ][o.e.n.Node               ] [stage] JVM arguments [-Des.networkaddress.cache.ttl=60, -Des.networkaddress.cache.negative.ttl=10, -Djava.security.manager=allow, -XX:+AlwaysPreTouch, -Xss1m, -Djava.awt.headless=true, -Dfile.encoding=UTF-8, -Djna.nosys=true, -XX:-OmitStackTraceInFastThrow, -Dio.netty.noUnsafe=true, -Dio.netty.noKeySetOptimization=true, -Dio.netty.recycler.maxCapacityPerThread=0, -Dlog4j.shutdownHookEnabled=false, -Dlog4j2.disable.jmx=true, -Dlog4j2.formatMsgNoLookups=true, -Djava.locale.providers=SPI,COMPAT, --add-opens=java.base/java.io=org.elasticsearch.preallocate, --add-opens=org.apache.lucene.core/org.apache.lucene.store=org.elasticsearch.vec, --enable-native-access=org.elasticsearch.nativeaccess, -XX:ReplayDataFile=/var/log/elasticsearch/replay_pid%p.log, -Djava.library.path=/usr/share/elasticsearch/lib/platform/linux-x64:/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib, -Djna.library.path=/usr/share/elasticsearch/lib/platform/linux-x64:/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib, -Des.distribution.type=deb, -XX:+UnlockDiagnosticVMOptions, -XX:G1NumCollectionsKeepPinned=10000000, -XX:+UseG1GC, -Djava.io.tmpdir=/tmp/elasticsearch-10079525150206968698, --add-modules=jdk.incubator.vector, -XX:+HeapDumpOnOutOfMemoryError, -XX:+ExitOnOutOfMemoryError, -XX:HeapDumpPath=/var/lib/elasticsearch, -XX:ErrorFile=/var/log/elasticsearch/hs_err_pid%p.log, -Xlog:gc*,gc+age=trace,safepoint:file=/var/log/elasticsearch/gc.log:utctime,level,pid,tags:filecount=32,filesize=64m, -Xms7805m, -Xmx7805m, -XX:MaxDirectMemorySize=4093640704, -XX:G1HeapRegionSize=4m, -XX:InitiatingHeapOccupancyPercent=30, -XX:G1ReservePercent=15, --module-path=/usr/share/elasticsearch/lib, --add-modules=jdk.net, --add-modules=ALL-MODULE-PATH, -Djdk.module.main=org.elasticsearch.server]
```

**Comment by sneakyx** [comment]
I could resolve this issue for now by changing the rights for the dir `/usr/lib/mysql/plugin/auth_pam_tool_dir`, but it doesn't look & feel correct to do this.

**Comment by chongshengdz** [comment]
yes, same issue elasticsearch cannot be started.

**Comment by chongshengdz** [comment]
AlmaLinux 9, the error is caused by
Caused by: java.io.UncheckedIOException: java.nio.file.AccessDeniedException: /usr/lib64/mysql/plugin/auth_pam_tool_dir

**Comment by Dazztee** [comment]
same issue here ubuntu 22 cannot start elastic , java.nio.file.AccessDeniedException: /usr/lib/mysql/plugin/auth_pam_tool_dir

**Comment by mousebat** [comment]
Same here - Ubuntu 24.04 LTS.

**Comment by kode54** [comment]
Same on Arch Linux using the AUR package to build it from source, which I doubt is a supported scenario, but I thought I'd mention it anyway.

**Comment by themew** [comment]
> I could resolve this issue for now by changing the rights for the dir `/usr/lib/mysql/plugin/auth_pam_tool_dir`, but it doesn't look & feel correct to do this.

Experiencing this with Debian 12.  Sorry, can you clarify what rights need to be changed so we can start Elasticsearch 8.14?

**Comment by sneakyx** [comment]
Because someone asked: I changed the right for this folder to 755. Then we could (re)start elasticsearch. But as I said, it doesn't feel right to give everyone rights to this folder.

**Comment by kode54** [comment]
That directory is already 755 on my system, and belongs to the mariadb package.

**Comment by gwbrown** [comment]
Hi, thanks for your bug report and interest in Elasticsearch! Fortunately I have good news: We have [already fixed this issue](https://github.com/elastic/elasticsearch/pull/108931) and [backported the fix](https://github.com/elastic/elasticsearch/pull/109419), which is expected to be released with 8.14.1. As such, I'm going to close this issue.

In the meantime, as noted, the workaround is to grant the Elasticsearch process read access to the directory or directories in question - there may be multiple, depending on your environment.

**Comment by sneakyx** [comment]
Hi @gwbrown ,
thanks for the fix. I can confirm: After updating elasticsearch to 8.14.1 and changing back the rights for the folder to 700, elasticsearch can be restarted without any errors.
Good work!
